src/test/java/com/hillel/items_exchange/controller/UserControllerTest.java

```diff
@@ -59,4 +94,162 @@ void negativeTestReceivingInformationAboutAnotherUser() throws Exception {
+        UserDto validUserDto = UserDtoUpdatingUtil.getValidUserDto();
+        validUserDto.setEmail("new.admin@gmail.com");
+        validUserDto.setFirstName("newFirstName");
+        validUserDto.setLastName("newLastName");
+
+        getResultActionsWithAuthenticationPrincipal(
+                HttpMethod.PUT, "/user/info/{id}", 1L, validUserDto, status().isAccepted())
+                .andDo(print())
+                .andExpect(jsonPath("$.email").value("new.admin@gmail.com"))
+                .andExpect(jsonPath("$.firstName").value("newFirstName"))
+                .andExpect(jsonPath("$.lastName").value("newLastName"));
+    }
+
+    @Test
+    @Transactional
+    @DataSet("database_init.yml")
+    @ExpectedDataSet(value = "user/update_child.yml", ignoreCols = {"last_online_time", "updated"})
+    void updateUserInfo_shouldUpdateUserChildren() throws Exception {
+        UserDto validUserDto = UserDtoUpdatingUtil.getValidUserDto();
+        validUserDto.getChildren().add(
+                new ChildDto(0, "female", LocalDate.of(2020, 1, 1)));
+
+        getResultActionsWithAuthenticationPrincipal(
+                HttpMethod.PUT, "/user/info/{id}", 1L, validUserDto, status().isAccepted())
+                .andDo(print())
+                .andExpect(jsonPath("$.children", hasSize(2)))
+                .andExpect(jsonPath("$.children[1].sex").value("female"));
+    }
+
+    @Test
+    @Transactional
+    @DataSet("database_init.yml")
+    void updateUserInfo_shouldReturnBadRequestIfInvalidChildBirthDate() throws Exception {
+        UserDto invalidUserDto = UserDtoUpdatingUtil.getInvalidUserDtoWithInvalidChildBirthDate();
+
+        getResultActionsWithAuthenticationPrincipal(
+                HttpMethod.PUT, "/user/info/{id}", 1L, invalidUserDto, status().isBadRequest()).andReturn();
+    }
+
+    @Test
+    @Transactional
+    @DataSet("database_init.yml")
+    void putUserDto_shouldBeThrownUnprocessableEntityException() throws Exception {
+        UserDto validUserDto = UserDtoUpdatingUtil.getValidUserDto();
+        validUserDto.getChildren().add(new ChildDto(5L, "male", LocalDate.now()));
+
+        MvcResult result = getResultActionsWithAuthenticationPrincipal(
+                HttpMethod.PUT, "/user/info/{id}", 1L, validUserDto, status().isUnprocessableEntity()).andReturn();
+        assertThat(result.getResolvedException(), is(instanceOf(UnprocessableEntityException.class)));
+    }
+
+    @Test
+    @Transactional
+    @DataSet("database_init.yml")
+    @ExpectedDataSet(value = "user/update_phone.yml", ignoreCols = {"last_online_time", "updated"})
+    void updateUserInfo_shouldUpdateUserPhones() throws Exception {
+        UserDto validUserDto = UserDtoUpdatingUtil.getValidUserDto();
+        validUserDto.getPhones().get(0).setPhoneNumber("+1 (234) 111-11-11");
+        validUserDto.getPhones().add(new PhoneDto(0, Boolean.FALSE, "+38.050.222.22.22", Boolean.FALSE));
```

Comment:
Maybe this can be simplified? Using list with not valid phones for example or something else, in order not to duplicate the code

Reply:
Firstly those phones are valid. And I did to show that masks of regexp are working good. But if somebody is agree with you, I will add them to List.